### PR TITLE
refactor: Consolidate new SpokePool/proxy deployment

### PR DESCRIPTION
**Pull request by pxrl**
This change aims to factor out quite a bit of duplicated logic.

For context, the "deployer" and the use of getNamedAccounts() was left out since it seems to be unused. hardhat.config.ts unconditionally calls getMnemonic(), which sources the private key from the environment, so the use of getNamedAccounts() may confuse and distract from the real address being used.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -23,16 +23,16 @@ jobs:
         id: release
         uses: manovotny/github-releases-for-automated-package-publishing-action@v2.0.1
 
-      # Perform installs, run tests, run a build step, etc. here, as needed.
-      - run: yarn
-
```
Comment (nicholaspai) [inline .github/workflows/publish.yml:28]: Is this a merge conflict change?

File: scripts/setupArbitrumSpokePool.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // @notice Logs ABI-encoded function data that can be relayed from HubPool to ArbitrumSpokePool to set it up.
 
-import { getContractFactory, ethers } from "../utils/utils";
+import { getContractFactory, ethers, hre } from "../utils/utils";
```
Comment (pxrl) [inline scripts/setupArbitrumSpokePool.ts:3]:
```suggestion
import { getContractFactory, ethers } from "../utils/utils";
```

File: utils/index.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
 export * from "./utils";
 export * from "./MerkleTree";
-export * from "./constants";
+export * from "./constants";
```
Comment (pxrl) [inline utils/index.ts:3]:
```suggestion
export * from "./constants";

```

File: .github/workflows/publish.yml (hunk 1 above)
Comment (pxrl) [reply .github/workflows/publish.yml]: Not sure, but it's not present in the PR any more. I've handled all the merge conflicts, and now the actual intended changes are all that remain.

**Review by nicholaspai — APPROVED** [review]

**Pull request merged**